hak3.sh

```diff
@@ -27,7 +27,7 @@ until [ -z "$1" ]; do
 done
 
 i=0
-ls -d */ 2> /dev/null | while read stud; do
+for stud in */; do
```

Comment:
If there are *no* subdirectories (not an unlikely scenario), this will execute the loop once (and try to mv "*/" .somedir). So if you make this change you should also use `shopt -s nullglob`.